src/storage/index.ts

```diff
@@ -50,16 +174,64 @@ export const saveAddvancedSettings = (settings: AdvancedAppSettings) => {
     "base64"
   );
 
-  localStorage.setItem(ADVANCED_SETTINGS_KET, encoded);
+  const str = localStorage.getItem(ADVANCED_SETTINGS_KET);
```

Comment:
export const saveAddvancedSettings = (settings: AdvancedAppSettings) => {
  const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString("base64");
  const str = localStorage.getItem(ADVANCED_SETTINGS_KET);
  const data = str ? JSON.parse(str) : [];

  // Check if the encoded settings already exist
  if (data.some((el: { encoded: string }) => el.encoded === encoded)) return;

  // Add new settings and save to local storage
  data.push({ encoded, active: data.length === 0, id: data.length + 1 });
  localStorage.setItem(ADVANCED_SETTINGS_KET, JSON.stringify(data));
};


could you rethink to use this code

Reply:
✔